src/NHibernate.Test/NHSpecificTest/GH2621Enum/Fixture.cs

```diff
@@ -0,0 +1,20 @@
+﻿using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH2621Enum
+{
+	public class Fixture : BugTestCase
+	{
+		[Test]
+		public void TestOk()
+		{
+			using (var s = OpenSession())
+			using (s.BeginTransaction())
+			{
+				var query = s.CreateQuery(
+					@"SELECT Name FROM NHibernate.Test.NHSpecificTest.GH2621Enum.ClassWithString ROOT WHERE ROOT.Kind = :kind");
+				query.SetParameter("kind", Kind.SomeKind);
```

Comment:
Why not using one of the following methods?

```
/* 1 */ query.SetParameter("kind", Kind.SomeKind, NHibernateUtil.Enum(typeof(Kind)));
/* 2 */ query.SetEnum("kind", Kind.SomeKind);
/* 3 */ query.SetParameter("kind", Kind.SomeKind.ToString());
```

Reply:
Because it's an existing code base where I cannot check all calls to SetParameter.

I would say that the change is a very specific one that does not hurt.